metaseq/model_parallel/criterions/vocab_parallel_cross_entropy.py

```diff
@@ -103,17 +107,9 @@ def reduce_metrics(logging_outputs) -> None:
         metrics.log_scalar(
             "loss", loss_sum / sample_size / math.log(2), sample_size, round=3
         )
-        if sample_size != ntokens:
-            metrics.log_scalar(
```

Comment:
Why did this get taken out?

Reply:
Hmm, so in the original code, sample_size is always equal to ntokens, so this is never executed and it doesn't change anything

In the new code, we aren't interested in nll_loss, it's some code from the MT days I think.